### PR TITLE
Profile picture added to Feature/admin users

### PR DESCRIPTION
**Pull request by KianMiridoozini**
User management rows now include the user's respective profile picture that links to the user's profile page

### CODE REVIEW

File: app/repositories/UserRepository.php [hunk 1]
```diff
@@ -228,24 +228,39 @@ public function searchUsers($query) {
...
                        WHEN u.account_status_id = 3 THEN 'Banned'
                        ELSE 'Unknown'
-                   END AS account_status_name
+                   END AS account_status_name,
+                   COALESCE(p.profile_picture, 'user-default.svg') AS profile_picture
             FROM users u
-            WHERE u.username LIKE :query
-        ");
-        $stmt->execute(['query' => '%' . $query . '%']);
+            LEFT JOIN user_profile p ON u.user_id = p.user_id
```
Comment (jayb2302) [inline app/repositories/UserRepository.php:242]: This function looks longer than it needs to be, is there a reason for that ?

File: app/views/admin/manage_users.php [hunk 2]
```diff
@@ -15,7 +15,9 @@
         <?php foreach ($users as $user): ?>
             <tr class="users-rows">
                 <td>
-                    <a href="/profile/<?= htmlspecialchars($user->getUsername()); ?>" class="view-profile">View Profile</a>
+                    <a href="/profile/<?= htmlspecialchars($user->getUsername()); ?>" class="view-profile">
+                        <img class="" alt="">
+                    </a>
```
Comment (jayb2302) [inline app/views/admin/manage_users.php:20]: no class and no alt

File: public/assets/js/admin.js [hunk 3]
```diff
@@ -10,14 +10,25 @@ document.addEventListener("DOMContentLoaded", () => {
...
-                <td><a href="/profile/${encodeURIComponent(user.username)}" class="view-profile">View Profile</a></td>
+                <td>
+                    <a href="/profile/${encodeURIComponent(user.username)}">
+                        <img src="${profileImage}" 
+                             alt="Profile picture of ${sanitizeHTML(user.username)}" 
+                             class="profile-pic" 
+                             width="40" 
+                             height="40">
+                    </a>
+                </td>
```
Comment (jayb2302) [inline public/assets/js/admin.js:29]: I see its here :P 
but we also have a class for user pictures, we should try to reuse classes

File: app/repositories/UserRepository.php (hunk 1 above)
Comment (KianMiridoozini) [reply app/repositories/UserRepository.php:242]: We wanted to implement javascript for some of the related functionality, this means some extra lifting from the php end, in this case the query used in the repository for this function.

This function is designed to fetch user data with optional filtering, handling account status labels and profile pictures in one query:

SQL Query: Selects user details, with:

CASE for readable account status ("Active," "Suspended," etc.).
COALESCE for default profile picture if missing.
LEFT JOIN to include profile picture data without excluding users without one.
Query Filtering: If a query (username search) is provided, a WHERE clause filters results to match the search term.

File: app/views/admin/manage_users.php (hunk 2 above)
Comment (KianMiridoozini) [reply app/views/admin/manage_users.php:20]: The class an alt are implemented through javascript.

As seen upon inspecting the page, profile pictures do have class and alt text.

File: public/assets/js/admin.js (hunk 3 above)
Comment (KianMiridoozini) [reply public/assets/js/admin.js:29]: I fail to find a class called user picture. The closest instance i found is  .control-user-picture

**Review by jayb2302 — APPROVED** [review]

**Pull request merged**